drivers/at86rf2xx/at86rf2xx_getset.c

```diff
@@ -328,93 +326,63 @@ void at86rf2xx_set_cca_threshold(at86rf2xx_t *dev, int8_t value)
     at86rf2xx_reg_write(dev, AT86RF2XX_REG__CCA_THRES, value);
 }
 
-void at86rf2xx_set_option(at86rf2xx_t *dev, uint16_t option, bool state)
+void at86rf2xx_set_option(at86rf2xx_t *dev,
+                          const uint16_t option, const bool state)
 {
     uint8_t tmp;
 
     DEBUG("set option %i to %i\n", option, state);
 
     /* set option field */
-    if (state) {
-        dev->netdev.flags |= option;
-        /* trigger option specific actions */
-        switch (option) {
-            case AT86RF2XX_OPT_CSMA:
+    (state) ? (dev->netdev.flags |=  option) :
+              (dev->netdev.flags &= ~option);
```

Comment:
WTF? can you please use `if` for this instead of black ternary magic ;-)?

Reply:
I bag to differ. If it were just a simple assignment, I would agree, but in this case I think

```C
if (state) {
    dev->netdev.flags |=  option;
}
else {
    dev->netdev.flags &= ~option;
}
```

looks much more intuitive than the statement above.

Reply:
IMO this is a rather subjective taste in coding style, you prefer if-else (which is fine) over ternary. However, I chose the latter because its consistent with the general coding style of the original driver, see L375ff in this file for instance. If that really is a blocker, I change, but then again should I change all other lines in question, too?

Reply:
> the latter because its consistent with the general coding style of the original driver, see L375ff in this file for instance

The lines in questions were introduced by you in this PR, at least according to the diff... The only usage of the ternary operator in this file in current master are in [l249](https://github.com/RIOT-OS/RIOT/blob/938ba0b3c5e1/drivers/at86rf2xx/at86rf2xx_getset.c#L249), [l267f](https://github.com/RIOT-OS/RIOT/blob/938ba0b3c5e1/drivers/at86rf2xx/at86rf2xx_getset.c#L267-L268), and [l279f](https://github.com/RIOT-OS/RIOT/blob/938ba0b3c5e1/drivers/at86rf2xx/at86rf2xx_getset.c#L279-L280). All of them are direct assignments! If you want a ternary operator so bad: I'm happy also happy with e.g.

```C
dev->netdev.flags = (state) ? (dev->netdev.flags | option) : (dev->netdev.flags & option);
```

It is true that it is a matter of opinion, but when it comes to readability, readability always comes first. The use of the ternary operator is to assign its result to something, which you do not do, which is just confusing (and, but I'm not sure of this also wasteful, since you are not using an assignment that is generated).

Reply:
oops, you're right it was me ... but now I remember why I did so 😉 

Actually my intention also was to improve readability because having an `if else` in every `case` statements makes it really ugly and I found the ternary variant more condensed and readable.